src/Common/RNGManager.cpp

```diff
@@ -0,0 +1,50 @@
+/*****************************************************************************
+ * Copyright (c) 2018-2022 openblack developers
+ *
+ * For a complete list of all authors, please refer to contributors.md
+ * Interested in contributing? Visit https://github.com/openblack/openblack
+ *
+ * openblack is licensed under the GNU General Public License version 3.
+ *****************************************************************************/
+
+#include "RNGManager.h"
+
+using namespace openblack;
+
+RNGManager& RNGManager::GetInstance()
+{
+	static RNGManager rngManager;
+	return rngManager;
+}
+
+uint32_t RNGManager::NextInt(uint32_t min, uint32_t max)
+{
+	static std::uniform_int_distribution<uint32_t> int_dist(min, max);
```

Comment:
This is not thread safe and it doesn't make sense to have it thread local either.